### ISSUE TITLE
web2.dav should be moved out to a separate project

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3072]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#3072|
|Type|task|
|Created|2008-03-09 19:29:28Z|

This will allow web2 trunk to make changes which break compatibility with web2.dav.  The idea is that it will let us speed up web2 development without having to refactor/rewrite all of the internally dependent dav code.

Presumably web2.dav should move to http://calendarserver.org.

Once web2 is folded back into twisted.web (i.e. done) we can reevaluate adding dav back in.

Attachments:

* [eradicate_dav.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/056/0569f0f363c447be6e9df64013c230115284ef91/3da58993cd25a815dcc4bb21800c4136a3e24666.diff) (302927 bytes) - added by dreid on 2008-03-16 01:08:31Z - A patch that eradicates DAV from trunk.

<details><summary>Searchable metadata</summary>

```
trac-id__3072 3072
type__task task
reporter__dreid dreid
priority__highest highest
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2_dav web2.dav
keywords__ 
time__1205090968000000 1205090968000000
changetime__1205773808000000 1205773808000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

This sounds like a great idea.  I feel like Twisted's development style has gummed up the works for the DAV project as well, and I've felt a little bad about not being able to speed up reviews and help with integrating patches and such.  I'm looking forward to streamlining processes for both projects, and the eventual re-integration of everything :).

Having a separate project on http://calendarserver.org will also provide some higher profile for Twisted, since the actual calendar server page doesn't currently mention Twisted, but presumably a "DAV support for Twisted" page would.

One thing that might be good to consider while doing this is having some web plugin mechanism so that web2.dav (or whatever its new name will be) can be easily deployed as a plugin via the twistd command line.  Maybe that's just a twistd plugin for dav, maybe it's something more.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez set status to `assigned`|
|-|-|

Here's what I propose:

 1. Merge the long-outstanding ACL branch (which is used by Calendar Server at calendarserver.org, web2.dav's most active user is using) down to trunk.
 1. Delete web2.dav from trunk.
 1. Create a branch for DAV, and add web2.dav there.

This enables web2 to continue development on trunk without being constrained by web2.dav.  It becomes the web2.dav developers' responsibility to reconcile any problems by merging forward periodically.  It also enables web2.dav to evolve a bit faster without affecting anything on trunk.

Some time after web2 is considered stable, web2.dav should be able to merge forward, become stable itself, then go back to review for a pull-down to trunk.

I would think it likely that a web2 release would happen without DAV before we do such a pull-down.

That advantage to this over pulling it out to another project is that we don't have to change the namespace out of twisted only to switch it back later, which is significant work on both web2.dav and its clients.

I talked to glyph about this yesterday, and he thought this would be a workable plan.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_4"></a>|@dreid commented|
|-|-|

(In [22825]) Merging forward

Refs [#3072](https://github.com/twisted/twisted/issues/3072)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez commented|
|-|-|

FWIW, attache patch is all removals except the last file (`twisted/web2/tap.py`).

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid set status to `closed`|
|-|-|

(In [22836]) Delete twisted.web2.dav and all references to it except in the autogenerated zsh completion files.

Author: dreid
Reviewer: exarkun

Refs [#3078](https://github.com/twisted/twisted/issues/3078)
Fixes [#3072](https://github.com/twisted/twisted/issues/3072)

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix commented|
|-|-|

(In [22867]) Delete twisted.web2.dav and all references to it except in the autogenerated zsh completion files.

Author: dreid
Reviewer: exarkun

Refs [#3078](https://github.com/twisted/twisted/issues/3078)
Fixes [#3072](https://github.com/twisted/twisted/issues/3072)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
